### PR TITLE
update metallb deployment to use pinned version

### PR DESCRIPTION
**Pull request by marcushines**

### CODE REVIEW

File: manifests/metallb/metallb.yaml [hunk 1]
```diff
@@ -364,7 +364,7 @@ spec:
             secretKeyRef:
               name: memberlist
               key: secretkey
-        image: quay.io/metallb/speaker:main
+        image: quay.io/metallb/speaker:v0.12.1
```
Comment (guoshiuan) [inline manifests/metallb/metallb.yaml:367]: Is this a temporary change to work around some issue?

**Comment by alexmasi** [comment]
I was just trying this out, about to send a PR. But then I tried main and I couldn't repro the issue. Did you see the issue? Also we should change both the speaker and controller to be pinned

File: manifests/metallb/metallb.yaml (hunk 1 above)
Comment (leomermelstein) [reply manifests/metallb/metallb.yaml:367]: metallib:main (apparently) doesn't accept '-config' anymore, breaking this deployment. We need to controller:main also needs to be changed to controller:v0.12.1 (at least until someone can sort out what happened to metallib)

Comment (marcushines) [reply manifests/metallb/metallb.yaml:367]: updated both images

**Comment by leomermelstein** [comment]
Please change controller:main to controller:v0.12.1 also, as that seems to be broken in the same way. Thanks.

**Review by alexmasi — APPROVED** [review]

**Review by guoshiuan — APPROVED** [review]

**Pull request merged**